packages/flutter_tools/lib/src/tester/flutter_tester.dart

```diff
@@ -235,8 +229,11 @@ class FlutterTesterDevice extends Device {
     ApplicationPackage app, {
     String userIdentifier,
   }) async {
-    _process?.kill();
-    _process = null;
+    if (_process != null) {
+      _process.kill();
+      await _process.exitCode;
```

Comment:
If we aren't going to use the exit code or the output of the subprocess for anything, there will be less risk of the tool hanging if it doesn't wait for the kill to work.